### ISSUE TITLE
Jira configurations: add proxy option

### ISSUE THREAD

**Issue by cedricbu** [DefectDojo/django-DefectDojo#9725]
It might be that some particular Jira instances can be accessible behind a Proxy only. Due to that it would be great if the Jira instances could get an proxy (and if possible with authentication), such that Defectdojo can communicate with it.

This Feature Request would be to add an optional Proxy (possibly proxy authentication as well) to Jira configurations.

Additional information:
After a quick look at the source, the JIRA() instantiator from the jira library can accept a proxy

**Comment by mtesauro** [comment]
It's been a couple of years since I did this but Python / Django supports `http_proxy` and `https_proxy` environmental variables so you can set those as ENV variables before starting DefectDojo. Here's a [decent article](https://www.delftstack.com/howto/python/set-http-proxy-settings-in-python/#use-export-to-set-http-proxy-settings-in-python) on that.

I'm assuming you want all outbound HTTP/HTTPS traffic going through a proxy - in your case the first and maybe only thing you need doing that is Jira.

HTH

**Comment by cedricbu** [comment]
No, that's the thing: only a single Jira instance is accessible only through a proxy. Any other connections (including other Jira instances) can go through direct network access.
But that's a good point: assuming that the proxy doesn't have any allow/deny list, it might be acceptable to enable the proxy for the entire Defectdojo instance
Thanks!

**Comment by valentijnscholten** [comment]
@cedricbu Are you able to test/implement this in a PR?

**Comment by cedricbu** [comment]
No, apologies, I don't have an instance to try out at the moment.
This feature request is probably not necessary: the workaround suggested by Matt works well and should be sufficient for the vast majority of cases.
